librecad/src/lib/modification/rs_modification.h

```diff
@@ -44,7 +44,8 @@ class RS_GraphicView;
 /**
  * Holds the data needed for move modifications.
  */
-struct RS_MoveData {
+class RS_MoveData {
+public:
```

Comment:
is this necessary?

Following C++ core guide lines: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-struct

Using struct is reasonable here, each member is logically independent and this no clear class/struct wide invariant

Reply:
Yes, the code needs to choose either a `class` or a `struct` for these and stick to it everywhere. As noted in the description, MSVC treats `class` and `struct` as different things in function signatures, so a function declaration in a header file that sees only the `class RS_MoveData;` forward declaration will not match the function definition in cpp file which sees the `struct` definition.
This actually had me headscratching for an hour, I couldn't find out why the linker was complaining about undefined symbols when "clearly" they were defined.

Reply:
Could you share some exact error messages to show the issue? Struct and class are always to distinct types in function signatures. There could be some incorrect forward declarations, which should be fixed instead.
The current design pattern of entity classes is by composite a class based data struct, which is POD, with no internal logic or invariant, therefore, as structs.
If you find error in linking to a class of such a data type, say, "class FooData". please search the source code for "class FooData", and replace all forward declarations to "struct FooData;". I do expect this would be the actual fix.

Reply:
Sure. First there's several warnings:
```
[...]\LibreCAD\librecad\src\lib\gui\rs_dialogfactoryinterface.h(55): warning C4099: 'RS_MoveData': type name first seen using 'struct' now seen using 'class'
[...]\LibreCAD\librecad\src\lib\modification\rs_modification.h(47): note: see declaration of 'RS_MoveData'
```
And then there's the linker error:
```
qc_dialogfactory.obj : error LNK2001: unresolved external symbol "public: virtual bool __cdecl QG_DialogFactory::requestMoveDialog(class RS_MoveData &)" (?requestMoveDialog@QG_DialogFactory@@UEAA_NAEAVRS_MoveData@@@Z)
..\..\windows\LibreCAD.exe : fatal error LNK1120: 1 unresolved externals
```
This results just from changing `class RS_MoveData` to `struct RS_MoveData`.

The problem is that there are forward declarations using `class`, while the definition is using `struct`. Based on what @dxli is saying, I believe the proper fix is to change the forward declarations to `struct`, rather than changing the definition to `class`; I'll change my PR to that if you agree.